dotnet/src/Connectors/Connectors.AzureAIInference.UnitTests/Services/AzureAIInferenceChatCompletionServiceTests.cs

```diff
@@ -0,0 +1,328 @@
+        Assert.Throws<ArgumentNullException>(() => new AzureAIInferenceChatCompletionService(endpoint: null, httpClient: this._httpClient));
+    }
+
+    [Fact]
+    public async Task ItGetChatMessageContentsShouldHaveModelIdDefinedAsync()
+    {
+        // Arrange
+        var chatCompletion = new AzureAIInferenceChatCompletionService(apiKey: "NOKEY", httpClient: this._httpClientWithBaseAddress);
+        this._messageHandlerStub.ResponseToReturn = new HttpResponseMessage(System.Net.HttpStatusCode.OK)
+        { Content = this.CreateDefaultStringContent() };
+
+        var chatHistory = new ChatHistory();
+        chatHistory.AddMessage(AuthorRole.User, "Hello");
+
+        // Act
+        var chatMessage = await chatCompletion.GetChatMessageContentAsync(chatHistory, this._executionSettings);
+
+        // Assert
+        Assert.NotNull(chatMessage.ModelId);
+        Assert.Equal("phi3-medium-4k", chatMessage.ModelId);
+    }
+
+    [Fact]
+    public async Task GetStreamingChatMessageContentsWorksCorrectlyAsync()
+    {
+        // Arrange
+        var service = new AzureAIInferenceChatCompletionService(httpClient: this._httpClientWithBaseAddress);
+        using var stream = File.OpenRead("TestData/chat_completion_streaming_response.txt");
+
+        this._messageHandlerStub.ResponseToReturn = new HttpResponseMessage(HttpStatusCode.OK)
+        {
+            Content = new StreamContent(stream)
+        };
+
+        // Act & Assert
+        var enumerator = service.GetStreamingChatMessageContentsAsync([]).GetAsyncEnumerator();
+
+        await enumerator.MoveNextAsync();
+        Assert.Equal(AuthorRole.Assistant, enumerator.Current.Role);
+
+        await enumerator.MoveNextAsync();
+        Assert.Equal("Test content", enumerator.Current.Content);
+        Assert.Equal("stop", enumerator.Current.Metadata?["FinishReason"]);
+    }
+
+    [Fact]
+    public async Task ItAddsSystemMessageAsync()
+    {
+        // Arrange
+        var chatCompletion = new AzureAIInferenceChatCompletionService(httpClient: this._httpClientWithBaseAddress);
+        this._messageHandlerStub.ResponseToReturn = new HttpResponseMessage(System.Net.HttpStatusCode.OK)
+        { Content = this.CreateDefaultStringContent() };
+        var chatHistory = new ChatHistory();
+        chatHistory.AddMessage(AuthorRole.User, "Hello");
+
+        // Act
+        await chatCompletion.GetChatMessageContentsAsync(chatHistory, this._executionSettings);
+
+        // Assert
+        var actualRequestContent = Encoding.UTF8.GetString(this._messageHandlerStub.RequestContent!);
+        Assert.NotNull(actualRequestContent);
+        var optionsJson = JsonSerializer.Deserialize<JsonElement>(actualRequestContent);
+
+        var messages = optionsJson.GetProperty("messages");
+        Assert.Equal(1, messages.GetArrayLength());
+
+        Assert.Equal("Hello", messages[0].GetProperty("content").GetString());
+        Assert.Equal("user", messages[0].GetProperty("role").GetString());
```

Comment:
The test title says it checks for system messages, but here it only checks for user messages.

Reply:
True, going to update the name, this connector has no `SystemMessage` in the settings.